qiita_db/test/test_archive.py

```diff
@@ -0,0 +1,67 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from unittest import TestCase, main
+
+from qiita_core.util import qiita_test_checker
+import qiita_db as qdb
+
+
+@qiita_test_checker()
+class ArchiveTest(TestCase):
+    def test_insert_from_biom_and_retrieve_feature_values(self):
+        # 1 - to test error as it's FASTQ
+        with self.assertRaises(ValueError) as err:
+            qdb.archive.Archive.insert_from_artifact(
+                qdb.artifact.Artifact(1), {})
+        self.assertEqual(
+            str(err.exception), 'To archive artifact must be BIOM but FASTQ')
+
+        # 7 - to test error due to not filepath biom
+        with self.assertRaises(ValueError) as err:
+            qdb.archive.Archive.insert_from_artifact(
+                qdb.artifact.Artifact(7), {})
+        self.assertEqual(
+            str(err.exception), 'The artifact has no biom files')
+
+        # testing specific artifacts and parameters
+        for i in [4, 5, 8, 9]:
+            qdb.archive.Archive.insert_from_artifact(
+                qdb.artifact.Artifact(i), {
+                    'featureA%d' % i: '{valuesA: vA, int: 1}',
+                    'featureB%d' % i: '{valuesB: vB, float: 1.1}'})
+
+        # now let's tests that all the inserts happen as expected
+        exp = {
+            'featureA4': '{valuesA: vA, int: 1}',
```

Comment:
are these supposed to be json? if so, they're not as the strings "vA" and "vB" are not quoted, nor are the keys

Reply:
well, they are strings but supposedly represent json, I can change for test if you feel strongly about it.

Reply:
if this is stuff generated from the code, then it certainly is not generating json. if this is supposed to be json, should the data contained in the insertion logic validate that the data are json?